src/panels/lovelace/cards/hui-entity-filter-card.ts

```diff
@@ -2,25 +2,37 @@ import { createCardElement } from "../common/create-card-element";
 import { processConfigEntities } from "../common/process-config-entities";
 import { LovelaceCard } from "../types";
 import { LovelaceCardConfig } from "../../../data/lovelace";
-import { EntityConfig } from "../entity-rows/types";
+import { EntityFilterEntityConfig } from "../entity-rows/types";
 import { HomeAssistant } from "../../../types";
 import { EntityFilterCardConfig } from "./types";
+import { evaluateFilter } from "../common/evaluate-filter";
 
 class EntityFilterCard extends HTMLElement implements LovelaceCard {
   public isPanel?: boolean;
   private _element?: LovelaceCard;
   private _config?: EntityFilterCardConfig;
-  private _configEntities?: EntityConfig[];
+  private _configEntities?: EntityFilterEntityConfig[];
   private _baseCardConfig?: LovelaceCardConfig;
   private _hass?: HomeAssistant;
-  private _oldEntities?: EntityConfig[];
+  private _oldEntities?: EntityFilterEntityConfig[];
 
   public getCardSize(): number {
     return this._element ? this._element.getCardSize() : 1;
   }
 
   public setConfig(config: EntityFilterCardConfig): void {
-    if (!config.state_filter || !Array.isArray(config.state_filter)) {
+    if (
+      (!config.state_filter &&
```

Comment:
Wouldn't this do it?
```
     if ((!config.state_filter || !Array.isArray(config.state_filter)) &&
        config.entities.filter((entity) => {
          return Boolean(typeof entity === "object" && entity.state_filter && Array.isArray(entity.state_filter);
        }).length !== config.entities.length)
```

Reply:
That would require that each entity have a state_filter

Reply:
No, it requires a default `state_filter` OR all entities have their own

Reply:
I see it now. Thanks, that's much cleaner!

Reply:
Even better to use `every`:
```
if (
    !(config.state_filter && Array.isArray(config.state_filter)) ||
    !config.entities.every((entity) => typeof entity === "object" && entity.state_filter && Array.isArray(entity.state_filter)
)
```

Reply:
@iantrich